spec.emu

```diff
@@ -136,189 +175,307 @@ location: https://github.com/tc39/proposal-decimal/
+          1. If _d2_ is *-∞*<sub>D</sub>, return Decimal128ValueToObject(*NaN*<sub>D</sub>).
+          1. Return Decimal128ValueToObject(*+∞*<sub>D</sub>).
+        1. If _d1_ is *-∞*<sub>D</sub>, then
+          1. If _d2_ is *+∞*<sub>D</sub>, return Decimal128ValueToObject(*NaN*<sub>D</sub>).
+          1. If _d2_ is *-∞*<sub>D</sub>, return Decimal128ValueToObject(*-∞*<sub>D</sub>).
+          1. Return Decimal128ValueToObject(*-∞*<sub>D</sub>).
+        1. Return Decimal128ValueToObject(« d1[0] / d2[0], min(d1[1], d2[1]) »).
+      </emu-alg>
     </emu-clause>
     <emu-clause id="sec-decimal.prototype.remainder">
       <h1>Decimal128.prototype.remainder ( _x_ )</h1>
-      <p>Computes the remainder of dividing *this* value by *_x_*. Exact division of *this* by *_x_* is carried out until the computation stops; the residue of that computation is returned. The sign of the result is the sign of *this* value. Returns a fresh result.</p>
+      <p>This method performs the following steps when called:</p>
+      <emu-alg>
+        1. Let _O_ be the **this** value.
+        1. Perform ? RequireInternalSlot(_O_, [[Decimal128Data]]).
+        1. Perform ? RequireInternalSlot(_x_, [[Decimal128Data]]).
+        1. Let _d1_ be _O_.[[Decimal128Data]].
+        1. Let _d2_ be _x_.[[Decimal128Data]].
+        1. If _d1_ or _d2_ is *NaN*<sub>D</sub>, return Decimal128ValueToObject(*NaN*<sub>D</sub>).
+        1. If _d1_ is *+∞*<sub>D</sub>, then
+          1. If _d2_ is *+∞*<sub>D</sub>, return Decimal128ValueToObject(*+∞*<sub>D</sub>).
+          1. If _d2_ is *-∞*<sub>D</sub>, return Decimal128ValueToObject(*NaN*<sub>D</sub>).
+          1. Return Decimal128ValueToObject(*+∞*<sub>D</sub>).
+        1. If _d1_ is *-∞*<sub>D</sub>, then
+          1. If _d2_ is *+∞*<sub>D</sub>, return Decimal128ValueToObject(*NaN*<sub>D</sub>).
+          1. If _d2_ is *-∞*<sub>D</sub>, return Decimal128ValueToObject(*-∞*<sub>D</sub>).
+          1. Return Decimal128ValueToObject(*-∞*<sub>D</sub>).
+        1. Return Decimal128ValueToObject(« d1[0] % d2[0], min(d1[1], d2[1]) »).
+      </emu-alg>
       <emu-note>
         <p>The remainder operation specified here deviates from the IEEE 754 remainder operation. We follow the <emu-xref href="#sec-numeric-types-number-remainder">the remainder operation for Number</emu-xref> rather than IEEE 754, which specifies that, for example, the remainder 46 and 10 is -4 rather than 6.</p>
       </emu-note>
     </emu-clause>
     <emu-clause id="sec-decimal.prototype.equals">
       <h1>Decimal128.prototype.equals ( _x_ )</h1>
-      <p>Compares the <emu-xref href="#dfn-decimal128-mathematical-value">mathematical value</emu-xref> of *this* value with the mathematical value of *_x_*, which is supposed to be a Decimal128 object. Returns *true* or *false*.</p>
+      <p>This method performs the following steps when called:</p>
       <emu-alg>
         1. If _x_ is not a Decimal128 object, then
```

Comment:
object or value? or is either fine?

Reply:
No, Decimal128 value isn't an ECMAScript language value, so _x_ can't be one. (But use RequireInternalSlot to determine whether _x_ is a Decimal128 object, rather than the "is not a" language.)

Reply:
I meant object in this case. I've adopted the use of RequireInternalSlot, rather than checking for being a Decimal128 object. Thanks!